### DISCUSSION TITLE
Add a WebSocket module in pyscript

### DISCUSSION THREAD

**Discussion by sansyrox** [pyscript/pyscript#1900] [Proposals]
Ideally, I would love to have an interface like

`from pyscript import WebSocket`

Meanwhile, I am also trying to create a Websocket connection using the JS module using

```
import js 
js.WebSocket()
```
But I am getting the following error

`pyodide.ffi.JsException: TypeError: Failed to construct 'WebSocket': Please use the 'new' operator, this DOM object constructor cannot be called as a function.`

How can I address this? And what does the community think about this proposal?

**Comment by WebReflection** [comment]
the convention for constructor (classes) in Pyodide and MicroPython is to use `.new(...args)` so that `js.WebSocket.new()` is what you are after?

**Reply by sansyrox** [reply]
Thank you @WebReflection 😄 

That does solve one part. However, it would be nice if we could the websocket api in the pyscript module directly
